### PR TITLE
Add support for linting .ts and .tsx files

### PR DESCRIPTION
**Pull request by LucianBuzzo**
Connects to #3

change-type: minor

### CODE REVIEW

File: lib/resin-lint.coffee [hunk 1]
```diff
@@ -44,35 +46,50 @@ parseJSON = (file) ->
...
+		source = read(file)
+		linter.lint(file, source, parsedConfig)
+	return linter.getResult()
+
+
 module.exports = (passed_params) ->
 	try
 		options = optimist(passed_params)
 			.usage('Usage: resin-lint [options] [...]')
 			.describe('f', 'Specify a coffeelint config file to override resin-lint rules')
```
Comment (thgreasi) [inline lib/resin-lint.coffee]: There are some occurrences of `coffee` in these `describe` statements.

File: lib/resin-lint.coffee [hunk 2]
```diff
@@ -7,8 +7,10 @@ merge = require('merge')
 optimist = require('optimist')
 coffeelint = require('coffeelint')
 reporter = require('coffeelint/lib/reporters/default')
+tslint = require('tslint')
 
-CONFIG_PATH = path.join(__dirname, '../config/coffeelint.json')
+CS_CONFIG_PATH = path.join(__dirname, '../config/coffeelint.json')
+TS_CONFIG_PATH = path.join(__dirname, '../config/tslint.json')
```
Comment (thgreasi) [inline lib/resin-lint.coffee]: Just an idea: How about using an object, containing the config filename (& possibly the target file types as well)?

File: lib/resin-lint.coffee [hunk 3]
```diff
@@ -96,29 +113,44 @@ module.exports = (passed_params) ->
...
-				quiet: false
-			report.publish()
-			process.on 'exit', ->
-				process.exit(errorReport.getExitCode())
+
+			if options.argv.typescript
+				scripts = findFiles(['ts', 'tsx'], paths)
+				errorReport = lintTsFiles(scripts, config)
+
+				console.log(errorReport.output)
```
[the remark this answers is not in the document]
Comment (LucianBuzzo) [reply lib/resin-lint.coffee]: Yep we need this, this is how we display the result of the linting process to the end user

**Comment by LucianBuzzo** [comment]
@thgreasi Thanks for the feedback. I've tightened up the code and I think it will be much better suited to adding more languages in the future

**Pull request merged**